src/main/java/com/theelixrlabs/healthcare/validation/Validator.java

```diff
@@ -58,4 +61,29 @@ public void validatePatientDTO(PatientDTO patientDTO) throws CustomException {
             throw new CustomException(messageUtil.getMessage(PatientConstants.INVALID_LAST_NAME_KEY));
         }
     }
+
+    /**
+     * Custom validation method for checking regex pattern of firstname lastname and Aadhaar number
+     *
+     * @param doctorDto : DoctorDto object containing doctor information.
+     * @throws CustomException : Class to handle custom exception
+     */
+    public void validateDoctor(DoctorDto doctorDto) throws CustomException {
+
+        if (doctorDto.getFirstName().isEmpty()) {
+            throw new CustomException(messageUtil.getMessage(MessageConstants.DOCTOR_FIRST_NAME_SHOULD_NOT_BE_EMPTY));
+        } else if (!(doctorDto.getFirstName().matches(DoctorConstants.ALPHA_CHARACTERS_REGEX))) {
+            throw new CustomException(messageUtil.getMessage(MessageConstants.DOCTOR_INVALID_FIRSTNAME));
+        }
+        if (doctorDto.getLastName().isEmpty()) {
+            throw new CustomException(messageUtil.getMessage(MessageConstants.DOCTOR_LAST_NAME_SHOULD_NOT_BE_EMPTY));
+        } else if (!(doctorDto.getLastName().matches(DoctorConstants.ALPHA_CHARACTERS_REGEX))) {
+            throw new CustomException(messageUtil.getMessage(MessageConstants.DOCTOR_INVALID_LASTNAME));
+        }
+        if (doctorDto.getAadhaarNumber().isEmpty()) {
+            throw new CustomException(messageUtil.getMessage(MessageConstants.DOCTOR_AADHAAR_NUMBER_SHOULD_NOT_BE_EMPTY));
+        } else if (!(doctorDto.getAadhaarNumber().matches(DoctorConstants.AADHAAR_REGEX_PATTERN))) {
+            throw new CustomException(messageUtil.getMessage(MessageConstants.DOCTOR_INVALID_AADHAAR_NUMBER));
+        }
```

Comment:
Do you actually need else if (...) ?
Why cant it be simply if block.

Reply:
Resolved.